File: src/java/magma/app/compile/lang/NumberFilter.java
package magma.app.compile.lang;

import magma.app.compile.rule.Filter;

import java.util.stream.IntStream;

public class NumberFilter implements Filter {
    private static boolean allDigits(String input) {
        return IntStream.range(0, input.length())
                .allMatch(i -> Character.isDigit(input.charAt(i)));
    }

    @Override
    public boolean filter(String input) {
        return input.startsWith("-")
                ? allDigits(input.substring(1))
                : allDigits(input);
    }
}

File: src/c/magma/app/rule/filter/NumberFilter.c
import java.util.function.Predicate;
import java.util.stream.IntStream;
struct NumberFilter implements Predicate<String> {
	 boolean allDigits(String input){
		return IntStream.range(0, input.length()).mapToObj(input.charAt).allMatch(Character.isDigit);
	}
	@Override
boolean test(String input){
		if(input.startsWith("-")){
			return allDigits(input.substring(1));
		}
		return allDigits(input);
	}
}

